### ISSUE TITLE
Wallet is not generating new addresses

### ISSUE THREAD

**Issue by BCREAM** [fyookball/electrum#138]
So I created a wallet and started receiving bcc. When the 20th address (the last one that is visible) received payment, electron cash generated a bunch of new addresses below it. I then received bcc into my 21st address.

However, after restarting my computer, my Electron Cash wallet is now stuck on the 20th address. Addresses 1-20 are all holding a balance, so it should automatically generate more addresses. The payment I received in my 21st address does not show up in the balance.

I am using Tails where everything gets deleted when the computer shuts down. Consequently, each time I start up Electron Cash I am essentially starting it up and restoring my wallet from scratch.

I guess it is a bug where it only generates new addresses when it actually sees the 20th address receiving coins? But when it has already been received and the wallet is being restored from scratch it doesnt generate any new ones past 20?

I was thinking it had to do something with the fact that I am using a Trezor with an out of date trezor library. However, I exported the master public key and created a view-only wallet, and the view-only wallet is still giving me the same trouble. Its only showing me 20 addresses, all with a blance, and not generating any new addresses.

Let me know if there is a way to fix this, or if I am doing something wrong. Thanks!

**Comment by BCREAM** [comment]
Hmmm just tried again for the fourth time and it seems to be working just fine now; I guess you can close this. Very strange.

**Comment by BCREAM** [comment]
Actually don't close this, it is back to not working again. I'll try to figure out what I did differently that one time to make it work.

**Comment by BCREAM** [comment]
I got it to work again, but not sure how. I think it has something to do with the order of events; it only workes if you open wallet, set proxy network settings, close wallet, connect to internet, reopen wallet.

Unfortunately I haven't been able to replicate this successful order since the second time it worked. However, I saved my wallet when it worked, so I shouldn't have this problem again. Very weird regardless, and should probably be looked into.

For reference I am using the latest Tails OS, and using the latest electron cash.
